### ISSUE TITLE
Set-AzureRmStorageAccount fails using -DisableEncryptionService

### ISSUE THREAD

**Issue by MiYanni** [Azure/azure-powershell#5243]
### Cmdlet(s)
Set-AzureRmStorageAccount
### PowerShell Version
5.1.16299.98
### Module Version
4.1.0
### OS Version
10.0.16299.98
### Description
Basically, I'm running the Test-StorageAccount ScenarioTest from StorageAccountTests.ps1. This section currently fails. It changes the Type, AccessTier, and EncryptionService of the storage account. It currently sends the same HTTP request as the TestStorageAccount.json SessionRecord. However, the API version is different, 2017-10-01 versus 2017-06-01 in the session record. That's the only difference I notice. But, it always fails stating, "The encryption key source missing from the request."
### Debug Output
PS C:\WINDOWS\system32> $stoSet = Set-AzureRmStorageAccount -ResourceGroupName rgtestme123 -Name storgtestme123 -Type 'Standard_LRS' -DisableEncryptionService 'Blob' -AccessTier 'Hot
' -Force
DEBUG: 10:44:59 AM - SetAzureStorageAccountCommand begin processing with ParameterSet 'StorageEncryption'.
DEBUG: 10:44:59 AM - using account id 'miyanni@microsoft.com'...
DEBUG: [Common.Authentication]: Authenticating using Account: 'miyanni@microsoft.com', environment: 'AzureCloud', tenant: '72f988bf-86f1-41af-91ab-2d7cd011db47'
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 01/08/2018 18:44:59: 32224205-ee60-4758-b758-a6b018e42c62 - AcquireTokenHandlerBase: === Token Acquisition started:
 Authority: https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/
 Resource: https://management.core.windows.net/
 ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
 CacheType: Microsoft.Azure.Commands.Common.Authentication.AuthenticationStoreTokenCache (2 items)
 Authentication Target: User

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 :
DEBUG: 01/08/2018 18:44:59: 32224205-ee60-4758-b758-a6b018e42c62 - TokenCache: Looking up cache for a token...
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 01/08/2018 18:44:59: 32224205-ee60-4758-b758-a6b018e42c62 - TokenCache: An item matching the requested resource was found in the cache
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 :
DEBUG: 01/08/2018 18:44:59: 32224205-ee60-4758-b758-a6b018e42c62 - TokenCache: 43.469300325 minutes left until token in cache expires
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 01/08/2018 18:44:59: 32224205-ee60-4758-b758-a6b018e42c62 - TokenCache: A matching item (access token or refresh token or both) was found in the cache
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 01/08/2018 18:44:59: 32224205-ee60-4758-b758-a6b018e42c62 - AcquireTokenHandlerBase: === Token Acquisition finished successfully. An access token was retuned:
 Access Token Hash: waRyHmZBwuWBVhifMlfW/Ts6bJrgdDJJH8yI0xoM6rc=
 Refresh Token Hash: mZjrmOYatZJkTUCj72EIO0iDzx56++VSR8sYUhabWoc=
 Expiration Time: 01/08/2018 19:28:27 +00:00
 User Hash: kxPty718soLw1l8balbeWriCjOQcjcR8IFUHLkFpPfQ=

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 01/08/2018 18:44:59:  - TokenCache: Serializing token cache with 2 items.
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 01/08/2018 18:44:59: 4ebb05ed-3d37-4485-9d2c-a9bbe7ebf56d - AcquireTokenHandlerBase: === Token Acquisition started:
 Authority: https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/
 Resource: https://management.core.windows.net/
 ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
 CacheType: Microsoft.Azure.Commands.Common.Authentication.AuthenticationStoreTokenCache (2 items)
 Authentication Target: User

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 :
DEBUG: 01/08/2018 18:44:59: 4ebb05ed-3d37-4485-9d2c-a9bbe7ebf56d - TokenCache: Looking up cache for a token...
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 01/08/2018 18:44:59: 4ebb05ed-3d37-4485-9d2c-a9bbe7ebf56d - TokenCache: An item matching the requested resource was found in the cache
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 :
DEBUG: 01/08/2018 18:44:59: 4ebb05ed-3d37-4485-9d2c-a9bbe7ebf56d - TokenCache: 43.4691664033333 minutes left until token in cache expires
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 01/08/2018 18:44:59: 4ebb05ed-3d37-4485-9d2c-a9bbe7ebf56d - TokenCache: A matching item (access token or refresh token or both) was found in the cache
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 01/08/2018 18:44:59: 4ebb05ed-3d37-4485-9d2c-a9bbe7ebf56d - AcquireTokenHandlerBase: === Token Acquisition finished successfully. An access token was retuned:
 Access Token Hash: waRyHmZBwuWBVhifMlfW/Ts6bJrgdDJJH8yI0xoM6rc=
 Refresh Token Hash: mZjrmOYatZJkTUCj72EIO0iDzx56++VSR8sYUhabWoc=
 Expiration Time: 01/08/2018 19:28:27 +00:00
 User Hash: kxPty718soLw1l8balbeWriCjOQcjcR8IFUHLkFpPfQ=

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 01/08/2018 18:44:59:  - TokenCache: Serializing token cache with 2 items.
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PATCH

Absolute Uri:
https://management.azure.com/subscriptions/c9cbd920-c00c-427c-852b-8aaf38badaeb/resourceGroups/rgtestme123/providers/Microsoft.Storage/storageAccounts/storgtestme123?api-version=201
7-10-01

Headers:
x-ms-client-request-id        : cf12a270-be29-4f10-aa38-80c7831359f7
accept-language               : en-US

Body:
{
  "sku": {
    "name": "Standard_LRS"
  },
  "properties": {
    "encryption": {
      "services": {
        "blob": {
          "enabled": false
        }
      }
    },
    "accessTier": "Hot"
  }
}

DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
BadRequest

Headers:
Pragma                        : no-cache
x-ms-request-id               : 64fbf80f-289d-4f40-99cc-854e49fc0916
Strict-Transport-Security     : max-age=31536000; includeSubDomains
Cache-Control                 : no-cache
Server                        : Microsoft-Azure-Storage-Resource-Provider/1.0,Microsoft-HTTPAPI/2.0
x-ms-ratelimit-remaining-subscription-writes: 1199
x-ms-correlation-request-id   : a2b63cb9-30ea-41c2-bc22-d5a3bea7b92b
x-ms-routing-request-id       : WESTUS2:20180108T184500Z:a2b63cb9-30ea-41c2-bc22-d5a3bea7b92b
Date                          : Mon, 08 Jan 2018 18:44:59 GMT

Body:
{
  "error": {
    "code": "MissingEncryptionKeySource",
    "message": "The encryption key source missing from the request."
  }
}

Set-AzureRmStorageAccount : The encryption key source missing from the request.
At line:1 char:11
+ $stoSet = Set-AzureRmStorageAccount -ResourceGroupName rgtestme123 -N ...
+           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Set-AzureRmStorageAccount], CloudException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Management.Storage.SetAzureStorageAccountCommand

DEBUG: 10:45:00 AM - SetAzureStorageAccountCommand end processing.
DEBUG: 10:45:00 AM - SetAzureStorageAccountCommand end processing.
### Script/Steps for Reproduction
Create a storage account with -EnableEncryptionService 'Blob,File' as part of New-AzureRmStorageAccount. Then, use -DisableEncryptionService 'Blob' with the Set-AzureRmStorageAccount. Observe that the command fails stating, "The encryption key source missing from the request."

**Comment by blueww** [comment]
@MiYanni 

Please add "-StorageEncryption" to the PowerShell cmdlet Set-AzureRMstorageAccount when use -DisableEncryptionService. 
You can see the usage in Example 6 in the cmdlet by run: `get-help Set-AzureRmStorageAccount -Examples`
This is caused by Server request Keysource whenever enable or disable Encryption. 
There are 2 kinds of Keysource: -KeyvaultEncryption and -KeyvaultEncryption. We can't set a default value, since it might overwrite the user setting on the server.

**Comment by MiYanni** [comment]
@blueww The -StorageEncryption parameter may allow Set-AzureRmStorageAccount to work now (on API version 2017-10-01), but this parameter was not required for API version 2017-06-01. So, it seems like a breaking change for the cmdlet, even though the cmdlet itself hasn't changed. The SessionRecords for the ScenarioTests do not use this -StorageEncryption parameter, but still pass. Running the tests live currently fail (because they don't have -StorageEncryption). Can you confirm this is a server-side issue caused by a new API version?

**Comment by blueww** [comment]
@MiYanni 

Yes, this is a server side behavior change caused the PowerShell behavior change. 
Actually, I have contact feature team for that, and the latest information I get is: Powershell should not support enabled/Disable Encryption anymore. So I plan to set the -DisableEncryptionService or -EnableEncryptionService to Obsolete soon, and remove them later.

**Comment by MiYanni** [comment]
@blueww I had to, as of yesterday, remove -StorageEncryption from the tests. They were actually failing when it was present. However, one of the encryption checks in the tests was still failing (after attempting to do -DisableEncryptionService). So, I removed that check for now. My PR for that should be picked up soon. Obviously, when Encryption is obsoleted, the tests will need to be fixed appropriately.

**Comment by blueww** [comment]
@MiYanni 

I have looked at your change in https://github.com/Azure/azure-powershell/pull/5301
Several comments are added.

**Comment by MiYanni** [comment]
I finally figured out what was the issue. The test recordings have not been re-recorded since API version 2016-12-01 and since then, the SDK was updated twice, once to 2017-06-01 and again to 2017-10-01. The tests need to be re-recorded every API change.

Currently, the TestController.cs was ignoring the API change. I have a branch where I've fixed the tests (including the Encryption issues and Location issues) and re-recorded all the tests. I will make a pull request for it tomorrow.
